src/main/java/net/revelc/code/impsort/ImpSort.java

```diff
@@ -73,9 +72,17 @@ public ImpSort(final Charset sourceEncoding, final Grouper grouper, final boolea
     this.lineEnding = lineEnding;
   }
 
-  public Result parseFile(final Path path) throws IOException {
-    String file = new String(Files.readAllBytes(path), sourceEncoding);
+  public Result parseFile(final Path path)
+      throws IOException, EmptyFileException, UnknownLineEndingException {
```

Comment:
This would be a good place to add javadoc, to explain the circumstances under which these new throws occur, especially since they are all `IOException` types (which... I'm not sure they need to be `IOException`... I don't think they represent an actual `IO` issue).

Reply:
I thought `IOException` was for things dealing with files. In general, these errors occur because of the file content.

It is possible to change these to have just `extends Throwable` instead, but it was already throwing an `IOException` on`parse()` already, unless you want a special exception for this case as well.

I am trying to find some similar example in the JDK itself. I am not sure right now which type of exception is best.

Reply:
Oh, definitely don't extend Throwable... Throwable includes Errors, and we want to be sure we're not creating a new custom category of Throwable by extending it directly. If anything, it'd be better to extend Exception or RuntimeException. I'm also fine with leaving it extend IOException on this one. We probably don't need to put that much thought into it. :smiley_cat:

Reply:
I think `RuntimeException` is not so good as it's unchecked. I am OK with changing it to `Exception` or just leaving it as it is now.

Reply:
Javadoc has been added to the `parseFile()` method.